src/server/tran_server.cpp

```diff
@@ -420,6 +414,8 @@ int
 tran_server::reset_main_connection ()
 {
   auto &conn_vec = m_page_server_conn_vec;
+  auto ulock = std::unique_lock<std::shared_mutex> { m_main_conn_mtx };
+
   /* the priority to select the main connection is the order in the container */
   const auto main_conn_cand = std::find_if (conn_vec.cbegin (), conn_vec.cend (),
```

Comment:
Nit.
```suggestion
  const auto main_conn_cand_it = std::find_if (conn_vec.cbegin (), conn_vec.cend (),
```
because it is an iterator.

Reply:
done.